### PR TITLE
Remove static

### PR DESCRIPTION
**Pull request by KostyaSha**
Relates to https://github.com/jenkinsci/build-pipeline-plugin/pull/64
Should unbreak many jenkinses

@patbos @daniel-beck

### CODE REVIEW

**Comment by KostyaSha** [comment]
Have no test cases because i hit many other unrelated bugs. With this fix i manually verified that builds appeared in history (excluding other unrelated bugs).

**Comment by oleg-nenashev** [comment]
It would be great to have a reference to Jenkins issue at least

**Comment by KostyaSha** [comment]
caused many issues, one of is referenced now

**Comment by patbos** [comment]
:+1:

**Comment by KostyaSha** [comment]
probably i found a build.xml that should deal exactly with static problem...

**Comment by oleg-nenashev** [comment]
@KostyaSha 
I'm not sure I'm ready to vote for this change.

Info from http://docs.oracle.com/javase/6/docs/platform/serialization/spec/serial-arch.html#4539 ...
//TL;DR: There will be a serialization of the parent <code>BuildPipelineView</code> to build.xml classes

> Note - Serialization of inner classes (i.e., nested classes that are not static member classes), including local and anonymous classes, is strongly discouraged for several reasons. Because inner classes declared in non-static contexts contain implicit non-transient references to enclosing class instances, serializing such an inner class instance will result in serialization of its associated outer class instance as well. Synthetic fields generated by javac (or other JavaTM compilers) to implement inner classes are implementation dependent and may vary between compilers; differences in such fields can disrupt compatibility as well as result in conflicting default serialVersionUID values. The names assigned to local and anonymous inner classes are also implementation dependent and may differ between compilers. Since inner classes cannot declare static members other than compile-time constant fields, they cannot use the serialPersistentFields mechanism to designate serializable fields. Finally, because inner classes associated with outer instances do not have zero-argument constructors (constructors of such inner classes implicitly accept the enclosing instance as a prepended parameter), they cannot implement Externalizable. None of the issues listed above, however, apply to static member classes.

**Comment by patbos** [comment]
Will it help to make MyUserIdCause not a inner class?

**Comment by KostyaSha** [comment]
> Will it help to make MyUserIdCause not a inner class?
> Would be nice if you could add a unitest with @LocalData with the build.xml.

This will break binary compatibility, xml references to inner class.
Yes, trying strip production build.xml to test.

**Comment by patbos** [comment]
Is it not possible to use  XSTREAM2.addCompatibilityAlias(String,Class) for that?
https://wiki.jenkins-ci.org/display/JENKINS/Hint+on+retaining+backward+compatibility

**Comment by KostyaSha** [comment]
> Is it not possible to use XSTREAM2.addCompatibilityAlias(String,Class) for that?
> https://wiki.jenkins-ci.org/display/JENKINS/Hint+on+retaining+backward+compatibility

I think in this case it should be enough to remove static without complicating code and adding xstream aliases

**Comment by KostyaSha** [comment]
No i unsure at all :(
@patbos feel free to fight

File: src/test/resources/au/com/centrumsystems/hudson/plugin/buildpipeline/BuildPipelineViewTest/testMyUserIdCauseConversionNoStatic/jobs/static-job/builds/2014-11-06_11-46-33/build.xml [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+                <permission>hudson.model.Run.Delete:user@user.com</permission>
+                <permission>hudson.model.Run.Update:user@user.com</permission>
+                <permission>hudson.model.Run.Update:user@user.com</permission>
+                <permission>hudson.model.View.Configure:user@user.com</permission>
+                <permission>hudson.model.View.Create:user@user.com</permission>
+                <permission>hudson.model.View.Delete:user@user.com</permission>
+                <permission>hudson.model.View.Read:user@user.com</permission>
+                <permission>hudson.scm.SCM.Tag:user@user.com</permission>
+                <!--<permission>org.jfrog.hudson.ArtifactoryPlugin.Promote:user@user.com</permission>-->
+                <!--<permission>org.jfrog.hudson.ArtifactoryPlugin.Release:user@user.com</permission>-->
```
Comment (KostyaSha) [inline src/test/resources/au/com/centrumsystems/hudson/plugin/buildpipeline/BuildPipelineViewTest/testMyUserIdCauseConversionNoStatic/jobs/static-job/builds/2014-11-06_11-46-33/build.xml]: Uncomment this and you will have broken jenkins that probably fixed by this https://github.com/jenkinsci/matrix-auth-plugin/pull/1

File: src/test/resources/au/com/centrumsystems/hudson/plugin/buildpipeline/BuildPipelineViewTest/testMyUserIdCauseConversionNoStatic/jobs/static-job/builds/2014-11-06_11-46-33/build.xml [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+          </outer-class>
+        </au.com.centrumsystems.hudson.plugin.buildpipeline.BuildPipelineView_-MyUserIdCause>
+      </causes>
+    </hudson.model.CauseAction>
+    <hudson.plugins.parameterizedtrigger.CapturedEnvironmentAction plugin="parameterized-trigger@2.17">
+      <env serialization="custom">
+        <unserializable-parents/>
+        <tree-map>
+          <default>
+            <comparator class="hudson.util.CaseInsensitiveComparator" reference="../../../../../org.jenkinsci.plugins.envinject.EnvInjectPluginAction/build/actions/hudson.plugins.parameterizedtrigger.CapturedEnvironmentAction/env/tree-map/default/comparator"/>
```
Comment (KostyaSha) [inline src/test/resources/au/com/centrumsystems/hudson/plugin/buildpipeline/BuildPipelineViewTest/testMyUserIdCauseConversionNoStatic/jobs/static-job/builds/2014-11-06_11-46-33/build.xml]: @jglick maybe you have any idea how restore old installations with this reference?
int and string becomes exposed to Action and Action casting fails

File: src/test/resources/au/com/centrumsystems/hudson/plugin/buildpipeline/BuildPipelineViewTest/testMyUserIdCauseConversionNoStatic/jobs/static-job/builds/2014-11-06_11-46-33/build.xml [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+          </default>
+          <!--Can't cast Integer to Action-->
+          <int>3</int>
```
Comment (KostyaSha) [inline src/test/resources/au/com/centrumsystems/hudson/plugin/buildpipeline/BuildPipelineViewTest/testMyUserIdCauseConversionNoStatic/jobs/static-job/builds/2014-11-06_11-46-33/build.xml]: ^highlighting place

**Comment by JTSweet** [comment]
@KostyaSha 
What is the current status of this pull request?

**Comment by KostyaSha** [comment]
@JTSweet we stuck, previous bad, serialized data can't be loaded. So everything works fine for new setups. Olds can't be migrated/cleaned automatically. I know only successful feedback where people manually cut crappy xml data from build.xml. 
AFAIR i asked something @patbos to look at...

**Pull request closed without merging**